File: Cc.Core/Models/Investment.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class Investment
    {
        public long Id { get; set; }
        public long BroadcastingId { get; set; }
        public DateTime DateFrom { get; set; }
        public DateTime? DateTo { get; set; }
        public string? Type { get; set; }
        public double Amount { get; set; }
        public string? Description { get; set; }
        public long? ProductId { get; set; }
        public string? InsertUser { get; set; }
        public DateTime? InsertDate { get; set; }
        public string? UpdateUser { get; set; }
        public DateTime? UpdateDate { get; set; }
        public string? DeleteUser { get; set; }
        public DateTime? DeleteDate { get; set; }
        public long? CategoryId { get; set; }

        public virtual Broadcasting Broadcasting { get; set; } = null!;
        public virtual Category? Category { get; set; }
        public virtual Product? Product { get; set; }
    }
}

File: Cc.Ui/src/models/investment.ts
import { Entity, resource, field, required, association } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';
import { Product } from './product';
import { cache } from '../app/cache';
import { LookupKey } from '../core/lookup';
import { Broadcasting } from './broadcasting';
import { Category } from './category';

@resource('investment')
export class Investment extends Entity {

    @field()
    id: number;
    @field()
    broadcastingId: number;
    @field()
    dateFrom: string;
    @field()
    dateTo: string;
    @field()
    type: string;
    @field()
    amount: number;
    @field()
    description: string;
    @field()
    productId: number;
    @field()
    categoryId: number;
    @field()
    insertUser: string;
    @field()
    insertDate: string;
    @field()
    updateUser: string;
    @field()
    updateDate: string;
    @field()
    deleteUser: string;
    @field()
    deleteDate: string;

    @association(Product, 'productId', (id: string | number) => cache(LookupKey.PRODUCT, id))
    product: Product;
    @association(Broadcasting, 'broadcastingId', (id: string | number) => cache(LookupKey.BROADCASTING, id))
    broadcasting: Broadcasting;
    @association(Category, 'categoryId', (id: string | number) => cache(LookupKey.CATEGORY, id))
    category: Category;
    
    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): Investment {

        let item = new Investment();
        return item;
    }
}

@resource('investment')
export class InvestmentSearchModel extends Entity implements Pagination {
    
    @field()
    broadcastingId?: number;
    @field()
    dateFrom?: string;
    @field()
    dateTo?: string;
    @field()
    type?: string;
    @field()
    amount?: number;
    @field()
    description?: string;
    @field()
    productId?: number;
    @field()
    deleteUser?: string;
    @field()
    deleteDate?: string;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}